File: Tugas 6/antrian.py
class Queue:

	def __init__(self, capacity):
		self.front = self.size = 0
		self.rear = capacity - 1
		self.Q = [None]*capacity
		self.capacity = capacity

	def isFull(self):
		return self.size == self.capacity

	def isEmpty(self):
		return self.size == 0

	def EnQueue(self, item):
		self.rear = (self.rear + 1) % (self.capacity)
		self.Q[self.rear] = item
		self.size = self.size + 1

	def DeQueue(self):
		self.front = (self.front + 1) % (self.capacity)
		self.size = self.size - 1

	def que_front(self):
		return self.Q[self.front]

	def que_rear(self):
		return self.Q[self.rear]

def main():
    print("")
    print("1. Ambil antrian")
    print("2. Panggil antrian")
    print("3. Keluar program")
    return int(input("Masukkan pilihan anda: "))

if __name__ == '__main__':
    queue = Queue(10)
    nomorAntrian = 0
    pilihan = 0

    print("Selamat datang di program Antrian Bank")
    print("========# oleh: Faturrohman Fairuz Zaki #========")
    
    while True:
        choice = main()
        print("")
        if choice == 1:
            if queue.isFull():
                print("Antrian sedang penuh!")
            else:
                nomorAntrian += 1
                queue.EnQueue(nomorAntrian)
                print("Antrian dengan nomor " + str(nomorAntrian) + " telah diambil.")
        elif choice == 2:
            if queue.isEmpty():
                print("Belum ada pengantre!")
            else:
                panggil = queue.que_front()
                queue.DeQueue()
                print("Antrian dengan nomor " + str(panggil) + " telah dipanggil.")
        elif choice == 3:
            print("Terima kasih telah menggunakan program ini!")
            break
        else:
            print("Input tidak valid!")
        
        antrian = queue.size
        print("Jumlah pengantre\t: " + str(antrian))
        if antrian > 0:
            print("Nasabah selanjuttnya\t: " + str(queue.que_front()))

File: Tugas 6/Antrian.java
import java.util.Scanner;

public class Antrian {

    int front, rear, size;
    int capacity;
    int array[];

    public Antrian(int capacity) {
        this.capacity = capacity;
        front = this.size = 0;
        rear = capacity - 1;
        array = new int[this.capacity];
    }

    boolean isFull(Antrian antrian) {
        return (antrian.size == antrian.capacity);
    }

    boolean isEmpty(Antrian antrian) {
        return (antrian.size == 0);
    }

    void enqueue(int item) {
        if (isFull(this))
            return;
        this.rear = (this.rear + 1) % this.capacity;
        this.array[this.rear] = item;
        this.size = this.size + 1;
    }

    int dequeue() {
        if (isEmpty(this))
            return Integer.MIN_VALUE;

        int item = this.array[this.front];
        this.front = (this.front + 1) % this.capacity;
        this.size = this.size - 1;
        return item;
    }

    int front() {
        if (isEmpty(this))
            return Integer.MIN_VALUE;

        return this.array[this.front];
    }

    int rear() {
        if (isEmpty(this))
            return Integer.MIN_VALUE;

        return this.array[this.rear];
    }

    public static void main(String[] args) {
        Antrian antrianBank = new Antrian(10);

        int nomorAntrian = 0;
        int pilihan = 0;
        Scanner input = new Scanner(System.in);

        System.out.println("Selamat datang di program Antrian Bank");
        System.out.println("oleh Faturrohman Fairuz Zaki");
        do {
            System.out.println("1. Ambil Antrian");
            System.out.println("2. Panggil Antrian");
            System.out.println("3. Keluar program");
            System.out.print("Masukkan pilihan anda : ");
            pilihan = input.nextInt();
            input.nextLine();
            System.out.println();

            switch (pilihan) {
                case 1:
                    if (antrianBank.isFull(antrianBank)) {
                        System.out.println("Antian sedang penuh!");
                        break;
                    } else {
                        nomorAntrian++;
                        antrianBank.enqueue(nomorAntrian);
                        System.out.println("Antrian dengan nomor " + nomorAntrian + " telah diambil");
                    }
                    break;
                case 2:
                    if (antrianBank.isEmpty(antrianBank)) {
                        System.out.println("Belum ada pengantre!");
                    } else {
                        int panggil = antrianBank.dequeue();
                        System.out.println("Antrian dengan nomor " + panggil + " telah dipanggil");
                    }
                    break;
                case 3:
                    System.out.println("Terima kasih telah menggunakan program ini!");
                    System.exit(0);
                default:
                    System.out.println("Input tidak valid!");
                    break;
            }
            System.out.println("Jumlah pengantre\t: " + antrianBank.size);
            if (antrianBank.size > 0){
                System.out.println("Nasabah selanjutnya\t: " + antrianBank.front());
            }
            System.out.println();
        } while (pilihan != 3);
        input.close();
    }
}